pkg/config/derive_test.go

```diff
@@ -0,0 +1,219 @@
+package config
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+
+	acsapi "github.com/openshift/openshift-azure/pkg/api"
+)
+
+func TestDerivedCloudProviderConf(t *testing.T) {
+	tests := []struct {
+		cs   api.OpenShiftManagedCluster
+		name string
+		want []byte
+	}{
+		{
+			name: "one",
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{AzProfile: &api.AzProfile{
+					TenantID:       "tenant",
+					SubscriptionID: "sub",
+					ResourceGroup:  "rg",
+				}, ServicePrincipalProfile: &api.ServicePrincipalProfile{
+					ClientID: "client_id",
+					Secret:   "client_secrett",
+				}},
+				Location: "eastus",
+			},
+			want: []byte(`aadClientId: client_id
+aadClientSecret: client_secrett
+location: eastus
+primaryScaleSetName: ss-compute
+resourceGroup: rg
+securityGroupName: nsg-compute
+subscriptionId: sub
+tenantId: tenant
+vmType: vmss
+`),
+		},
+	}
+
+	d := derived{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := d.CloudProviderConf(&tt.cs)
+			if err != nil {
+				t.Errorf("derived.CloudProviderConf() error = %v", err)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("derived.CloudProviderConf() = \"%v\", want \"%v\"", string(got), string(tt.want))
+			}
+		})
+	}
+}
+
+func TestDerivedMasterLBCNamePrefix(t *testing.T) {
+	tests := []struct {
```

Comment:
20 lines of boilerplate is not appropriate for a function which tests strings.Split, please simplify (comment probably applies to all tests in this file where the test matrix length is 1 and it is highly unlikely to increase)